arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -947,3 +950,47 @@ void MaintenanceFeature::proceed() {
   _pauseUntil = std::chrono::steady_clock::duration::zero();
 }
 
+bool MaintenanceFeature::isShardLocked(ShardID shardId,
```

Comment:
And maybe return either `nullptr` or `it->second`?

Reply:
I could return a pair, `std::optional` is off limits, since this must be ported back to 3.5. I could return `nullptr` if it is not found, the good old C convention... :-)

Reply:
You could fall back to `boost::optional` for the 3.5 backport. ;)

I wouldn't object to just returning either a valid pointer or `nullptr` either.

Anyway, changing the style of returning is less important than the name.